### PR TITLE
Tour managment/tm 4 transportation list

### PR DESCRIPTION
**Pull request by mirztate1-flutterint**

### CODE REVIEW

File: src/src_V2/modules/manager/supplier-management/accommodation/dialogs/AllFilters.dialog.jsx [hunk 1]
```diff
@@ -60,58 +67,68 @@ const BootstrapDialogTitle = (props) => {
...
+  const autoCompleteChangeHandler = (type) => (e, value) => setFieldValue(type, value);
+  const formikData = {
+    validationSchema: AccommodationFilterSchema(),
+    initialValues: FilterInitialValues(),
+  };
+
+  const { values, errors, touched, handleBlur, handleChange, setFieldValue, setValues } =
+    useFormik(formikData);
+
+  const handleReset = () => {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/accommodation/dialogs/AllFilters.dialog.jsx:45]: I think we can use the `FilterInitialValues` in handleReset, how do you think?

File: src/src_V2/modules/manager/supplier-management/food/add/Menu.jsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { Languages } from '../../transportation/constants';
 import { menuInitialValues } from '../../../../../utils/schemas/tourManagment/food';
 import AddAttachment from '../../components/add-attachment/AddAttachment';
 
-export default function Menu({ parentRef }) {
+export default function Guides({ parentRef }) {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/food/add/Menu.jsx]: Why we need this change?

File: src/src_V2/modules/manager/supplier-management/food/add/Menu.jsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ export default function Menu({ parentRef }) {
                 <TextField
                   {...params}
                   name='menuTags'
-                  label='Menu Tags'
+                  label='Guides Tags'
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/food/add/Menu.jsx]: the same for this one. ?

File: src/src_V2/modules/manager/supplier-management/food/dialogs/FoodDetails/Menu.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          variant='contained'
+          key={menu.id}
+          className={styles.detailsServiceBtns}
+          component='span'
+        >
+          {menu.label}
+        </Button>
+      ))}
+      <Typography className={styles.roomTitle}>MENU / ATTACHMENTS</Typography>
+      {['Special Guides 2021.pdf', 'Drinks.pdf'].map((attachment) => (
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/food/dialogs/FoodDetails/Menu.jsx]: please move this mock to constant.js

File: src/src_V2/modules/manager/supplier-management/guide/index.jsx [hunk 5]
```diff
@@ -2,6 +2,7 @@ import { Route, Routes } from 'react-router-dom';
 
 import AddGuide from './add/Index';
 import GudeNoData from './GuideNoData';
+import FoodList from './list/Index';
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/guide/index.jsx]: please change to Guide

File: src/src_V2/modules/manager/supplier-management/guide/index.jsx [hunk 6]
```diff
@@ -15,6 +16,7 @@ function routeConfigs() {
   return [
     { index: true, element: <GudeNoData /> },
     { path: ROUTES.ADD, element: <AddGuide /> },
+    { path: ROUTES.LIST, element: <FoodList /> },
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/guide/index.jsx]: the same

File: src/src_V2/modules/manager/supplier-management/transportation/add/Drivers.jsx [hunk 7]
```diff
@@ -6,7 +6,7 @@ import { AddCard, InfoCard } from '../../components';
 
 import styles from './style.module.css';
 
-export default function Drivers({ parentRef }) {
+export default function Rooms({ parentRef }) {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/add/Drivers.jsx]: Why we need this change?

File: src/src_V2/modules/manager/supplier-management/transportation/add/Drivers.jsx [hunk 8]
```diff
@@ -70,7 +70,7 @@ export default function Drivers({ parentRef }) {
     <Box className={styles.content}>
       <Grid container spacing={1}>
         <AddCard
-          title='Drivers'
+          title='Rooms'
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/add/Drivers.jsx]: the same comment here

File: src/src_V2/modules/manager/supplier-management/transportation/dialogs/AllFilters.dialog.jsx [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+  const formikData = {
+    validationSchema: TransportationFiltersSchema(),
+    initialValues: FilterInitialValues(),
+  };
+
+  const { values, errors, touched, handleBlur, handleChange, setFieldValue, setValues } =
+    useFormik(formikData);
+
+  const handleReset = () => {
+    setValues({
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/dialogs/AllFilters.dialog.jsx]: use initial values here

File: src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Cars.jsx [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+import { Grid } from '@mui/material';
+import { InfoCard } from '../../../components';
+
+import styles from '../style.module.css';
+import DialogManager from '../Index';
+
+export default function Drivers() {
+  const [dialogManagerState, onShowHideDialog] = useState({ open: false });
+  const [cars, setCars] = useState([
+    {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Cars.jsx]: please move to constants.js

File: src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Drivers.jsx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+import { InfoCard } from '../../../components';
+
+import styles from '../style.module.css';
+import DialogManager from '../Index';
+
+export default function Drivers() {
+  const [dialogManagerState, onShowHideDialog] = useState({ open: false });
+  const [drivers, setDrivers] = useState([
+    {
+      id: 1,
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Drivers.jsx]: move to constans.js

File: src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/TransportationDetails.dialog.jsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        className={`${styles.container} ${styles.header}`}
+      >
+        Agency / Agency Name Example
+      </BootstrapDialogTitle>
+      <DialogContent dividers className={styles.container}>
+        <Box className={styles.dialogMenu}>
+          <TabContext value={value}>
+            <Box
+              className={styles.headerMenu}
+              sx={{ borderBottom: 1, borderColor: 'divider' }}
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/TransportationDetails.dialog.jsx]: need we inline(sx) styling here? can't we use className

File: src/src_V2/modules/manager/supplier-management/components/bootstrapDialogTitle/BootstrapDialogTitle.jsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  '& .MuiDialogActions-root': {
+    padding: theme.spacing(1),
+  },
+}));
+
+export const BootstrapDialogTitle = (props) => {
+  const { children, onClose, ...other } = props;
+
+  return (
+    <DialogTitle sx={{ m: 0, p: 2 }} {...other}>
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/components/bootstrapDialogTitle/BootstrapDialogTitle.jsx]: inline styling

File: src/src_V2/modules/manager/supplier-management/components/bootstrapDialogTitle/BootstrapDialogTitle.jsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
+import { Dialog, DialogTitle, IconButton } from '@mui/material';
+import { Close, Delete, Edit } from '@mui/icons-material';
+import styles from '../../food/dialogs/style.module.css';
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/components/bootstrapDialogTitle/BootstrapDialogTitle.jsx]: please reorder imports

File: src/src_V2/modules/manager/supplier-management/food/dialogs/AllFilters.dialog.jsx [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+    padding: theme.spacing(2),
+    paddingTop: 30,
+    paddingBottom: 50,
+  },
+  '& .MuiDialogActions-root': {
+    padding: theme.spacing(1),
+  },
+}));
+
+const BootstrapDialogTitle = (props) => {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/food/dialogs/AllFilters.dialog.jsx]: You already have made  sharable this component, please resue it here.

File: src/src_V2/modules/manager/supplier-management/food/dialogs/AllFilters.dialog.jsx [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+            </Grid>
+          </Grid>
+          <Grid item container spacing={3}>
+            <Grid item xs={2.6}>
+              <FormControlLabel control={<Checkbox name='gilad' />} label='Weekend' />
+            </Grid>
+            <Grid item xs={3.2}>
+              <TextField
+                fullWidth
+                type='time'
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/food/dialogs/AllFilters.dialog.jsx:110]: The field have the same properties, make them unique

File: src/src_V2/modules/manager/supplier-management/food/list/Menu.jsx [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <div>
+      <MoreVert className={styles.moreVertIcon} onClick={handleClick} />
+      <StyledMenu
+        id='demo-customized-menu'
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/food/list/Menu.jsx]: remove additional props

File: src/src_V2/modules/manager/supplier-management/guide/dialogs/AllFilters.dialog.jsx [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+    padding: theme.spacing(2),
+    paddingTop: 30,
+    paddingBottom: 50,
+  },
+  '& .MuiDialogActions-root': {
+    padding: theme.spacing(1),
+  },
+}));
+
+const BootstrapDialogTitle = (props) => {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/guide/dialogs/AllFilters.dialog.jsx]: use sharable one

**Review by igortokhunts — CHANGES_REQUESTED** [review]

File: src/src_V2/modules/manager/supplier-management/accommodation/dialogs/AllFilters.dialog.jsx (hunk 1 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/accommodation/dialogs/AllFilters.dialog.jsx:45]: done!

File: src/src_V2/modules/manager/supplier-management/food/add/Menu.jsx (hunk 3 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/food/add/Menu.jsx]: I have these type of changes after merging the dev branch with mine. 
but fixed it!

File: src/src_V2/modules/manager/supplier-management/food/dialogs/FoodDetails/Menu.jsx (hunk 4 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/food/dialogs/FoodDetails/Menu.jsx]: done!

File: src/src_V2/modules/manager/supplier-management/food/dialogs/AllFilters.dialog.jsx (hunk 15 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/food/dialogs/AllFilters.dialog.jsx]: done!

File: src/src_V2/modules/manager/config.js [hunk 19]
```diff
@@ -56,7 +56,7 @@ export const managerSidebarConfig = [
         path: '/manager/activity',
       },
       {
-        title: 'Guides',
+        title: 'Activities',
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/config.js]: Please revert this

File: src/src_V2/modules/manager/supplier-management/activity/dialogs/Index.jsx [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+export default function DialogManager({ data, onShowHideDialog }) {
+  function onClose() {
+    onShowHideDialog({
+      mode: '',
+      open: false,
+    });
+  }
+
+  return (
+    <Dialog open={data.open} onClose={onClose} fullWidth PaperProps={{ width: 750 }}>
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/activity/dialogs/Index.jsx]: nned we set this `fullWidth PaperProps={{ width: 750 }}` ?

File: src/src_V2/modules/manager/supplier-management/activity/list/FilterBlock.jsx [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+import styles from './style.module.css';
+
+export default function FilterBlock() {
+  const [dialogManagerState, onShowHideDialog] = useState({ open: false });
+
+  return (
+    <Box className={styles.box}>
+      <div className={styles.leftSection}>
+        <Typography className={styles.title}>Activities</Typography>
+        <Grid container rowSpacing={3} spacing={2}>
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/activity/list/FilterBlock.jsx:41]: don't we need validation/initialValues here?

File: src/src_V2/modules/manager/supplier-management/activity/list/Index.jsx [hunk 22]
```diff
@@ -0,0 +1,146 @@
...
+  TableBody,
+  TableCell,
+  TableContainer,
+  TablePagination,
+} from '@mui/material';
+import FilterBlock from './FilterBlock';
+import { HeadCells, TableRows } from '../constants';
+import DialogManager from '../dialogs/Index';
+import TableMenuActions from '../../../supplier-management/components/tableVerticalActionsMenu/Menu';
+
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/activity/list/Index.jsx]: spacing ?

File: src/src_V2/modules/manager/supplier-management/activity/list/Index.jsx [hunk 23]
```diff
@@ -0,0 +1,146 @@
...
+      mode: 'activityDetails',
+      actions: {},
+    });
+  }
+
+  return (
+    <>
+      <FilterBlock />
+      <Box className={styles.table}>
+        <Paper sx={{ width: '100%', mb: 2 }}>
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/activity/list/Index.jsx]: inline styles

File: src/src_V2/modules/manager/supplier-management/guide/add/Guide.jsx [hunk 24]
```diff
@@ -64,13 +64,11 @@ export default function Guide({ parentRef }) {
...
   useEffect(addGuidesToTransportationData, [guides]);
 
-  console.log(guides);
-
   return (
     <Box className={styles.content}>
       <Grid container spacing={1}>
         <AddCard
-          title='Guides'
+          title='Attributes'
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/guide/add/Guide.jsx]: ???

File: src/src_V2/modules/manager/supplier-management/guide/list/FilterBlock.jsx [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+  function openAllFiltersDialog() {
+    onShowHideDialog({
+      open: true,
+      mode: 'guideFilters',
+      actions: {},
+    });
+  }
+
+  return (
+    <Box className={styles.box}>
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/guide/list/FilterBlock.jsx:36]: don't need validation/initialValues here?

**Review by igortokhunts — CHANGES_REQUESTED** [review]

File: src/src_V2/modules/manager/supplier-management/guide/add/Guide.jsx (hunk 24 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/guide/add/Guide.jsx]: changed

File: src/src_V2/modules/manager/supplier-management/activity/dialogs/Index.jsx (hunk 20 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/activity/dialogs/Index.jsx]: removed it

File: src/src_V2/modules/manager/supplier-management/activity/list/Index.jsx (hunk 23 above)
Comment (mirztate1-flutterint) [reply src/src_V2/modules/manager/supplier-management/activity/list/Index.jsx]: changed

File: src/src_V2/modules/manager/supplier-management/components/bootstrapDialogTitle/BootstrapDialogTitle.jsx [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+import PropTypes from 'prop-types';
+
+import styles from './style.module.css';
+
+/*TODO see if we can merge this with BootstrapDialog in
+AllFilter file, otherwise get this out from components folder*/
+
+export const BootstrapDialog = styled(Dialog)(({ styles }) => {
+  return styles;
+  // return {
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/components/bootstrapDialogTitle/BootstrapDialogTitle.jsx]: comment

File: src/src_V2/modules/manager/supplier-management/components/dialogs/Details.dialog.jsx [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+        Agency / Agency Name Example
+      </BootstrapDialogTitle>
+      <DialogContent dividers className={styles.container}>
+        <Box className={styles.dialogMenu}>
+          <TabContext value={value}>
+            <Box
+              className={styles.headerMenu}
+              sx={{ borderBottom: 1, borderColor: 'divider' }}
+            >
+              <TabList onChange={handleChange}>
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/components/dialogs/Details.dialog.jsx:34]: map

File: src/src_V2/modules/manager/supplier-management/guide/dialogs/ViewGuide.dialog.jsx [hunk 28]
```diff
@@ -71,14 +73,14 @@ export default function ViewRoomDialog({ data: { state: guide, actions } }) {
           </Grid>
           <Grid item xs={12}>
             <Typography>Languages</Typography>
-            {guide.languages.map((item) => (
+            {guide.languages?.map((item) => (
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/guide/dialogs/ViewGuide.dialog.jsx:76]: remove ?

File: src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Cars.jsx [hunk 29]
```diff
@@ -0,0 +1,85 @@
...
+      actions: editRoom,
+      state: cars.find((driver) => driver.id === id),
+    });
+  }
+
+  function openViewCardDialog(id) {
+    onShowHideDialog({
+      open: true,
+      mode: 'view-car',
+      state: cars.find((driver) => driver.id === id),
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Cars.jsx:46]: naming

File: src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Drivers.jsx [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+      mode: 'delete',
+      actions: deleteRoom,
+    });
+  }
+
+  function openEditCardDialog(id) {
+    onShowHideDialog({
+      open: true,
+      mode: 'edit-driver',
+      actions: editRoom,
```
Comment (igortokhunts) [inline src/src_V2/modules/manager/supplier-management/transportation/dialogs/TransportationDetails/Drivers.jsx:39]: naming

File: src/src_V2/utils/schemas/tourManagment/transportation.js [hunk 31]
```diff
@@ -75,3 +90,25 @@ export function addCarSchema() {
     model: BaseSchemas.requiredText(50),
   });
 }
+
+export function partnershipSchema() {
```
Comment (igortokhunts) [inline src/src_V2/utils/schemas/tourManagment/transportation.js:94]: move to components.js

**Comment by mirztate1-flutterint** [comment]
Some issues, which were resolved in TM-11

**Pull request closed without merging**